### PR TITLE
Run & Deploy Changes for the Turbine-core Implementation on Turbine-PY

### PR DESCRIPTION
**Pull request by anna-cross**
## Description of change

CLI changes for turbine-core implementation on turbine-py, works along with changes made to https://github.com/meroxa/turbine-py/pull/219

Fixes: 
https://github.com/meroxa/cli/issues/694
https://github.com/meroxa/cli/issues/693

## Type of change

- [ ]  New feature
- [ ]  Bug fix
- [x]  Refactor
- [ ]  Documentation

## How was this tested?

To test the following code with run command - 

1.  Grab latest changes to turbine-py and modify setup.config to point to pkg/ code (turbine-core implementation) instead of src/ code (old code) 
2. Run - `pip install -U turbine-py/ ` on your turbine-py
3. You should be able to now execute new core commands, such as `run` 
4. Install these cli changes with 'make install' on this branch 
5. Create a data app and run "meroxa app run ." 

To test the following code with deploy command - 

If this `[PR](https://github.com/meroxa/turbine-py/pull/219)` isnt merged at the time of testing and  this [PR](https://github.com/meroxa/turbine-core/pull/128) isnt merged at the time of testing, grab them both and do the following: 

We need to basically force the data app to use this version of turbine-py, instead of pip grabbing latests published version 
1. In your turbine-py repo, find requirements.txt and comment out any turbine-py requirements 
2. Modify Dockerfile to` pip install install -U turbine-py/`
3.  Run - `pip install -U turbine-py/ ` on your turbine-py 
4.  Run 'meroxa app init app --lang py' to create a new app 
5. The app generated should match the template from https://github.com/meroxa/turbine-core/pull/128 
6. Then grab the entire folder of turbine-py repo and drop it into your data app 
7. Then run 'meroxa app deploy .' 

- [ ]  Unit Tests
- [x]  Tested in staging
- [ ]  Tested in minikube

## Demo

Slack - https://meroxa.slack.com/archives/C03F15EKFU3/p1682636513287819

### CODE REVIEW

File: cmd/meroxa/turbine/python/cli.go [hunk 1]
```diff
@@ -3,14 +3,24 @@ package turbinepy
...
+	runner            server.Server
+	builder           server.Server
 }
 
 func New(l log.Logger, appPath string) turbine.CLI {
-	return &turbinePyCLI{logger: l, appPath: appPath}
+	return &turbinePyCLI{
+		logger:            l,
+		appPath:           appPath,
+		grpcListenAddress: server.ListenAddress,
```
Comment (dianadoherty) [inline cmd/meroxa/turbine/python/cli.go:22]: You might need to rebase your PR off of Lyu's run, so you can include things like:
```suggestion
		grpcListenAddress: internal.RandomLocalAddr(),,
```

**Review by dianadoherty — COMMENTED** [review]
Looks good, aside from small suggestion

File: cmd/meroxa/turbine/python/cli.go (hunk 1 above)
Comment (lyuboxa) [reply cmd/meroxa/turbine/python/cli.go:22]: You need these, otherwise all the runners will be clashing

Comment (lyuboxa) [reply cmd/meroxa/turbine/python/cli.go:22]: @dianadoherty there is an extra `,`

Comment (anna-cross) [reply cmd/meroxa/turbine/python/cli.go:22]: @lyuboxa Just to clarify, do you mean we should be using server.ListenAddress for grpcListenAddress? Or is internal.RandomLocalAddr() ok to use for it?

File: cmd/meroxa/turbine/python/internal/cmd.go [hunk 2]
```diff
@@ -0,0 +1,41 @@
...
+func RunTurbineCmd(appPath string, command TurbineCommand, env map[string]string, flags ...string) (string, error) {
+	cmd := NewTurbineCmd(appPath, command, env, flags...)
+	cmd.Stderr = nil
+	cmd.Stdout = nil
+	cmd.Dir = appPath
+	out, err := cmd.CombinedOutput()
+	return strings.TrimSpace(string(out)), err
+}
+
+func NewTurbineCmd(appPath string, command TurbineCommand, env map[string]string, flags ...string) *exec.Cmd {
```
Comment (dianadoherty) [inline cmd/meroxa/turbine/python/internal/cmd.go:28]:
```suggestion
func NewTurbineCmd(appPath string, env map[string]string, flags ...string) *exec.Cmd {
```

File: cmd/meroxa/turbine/python/internal/cmd.go [hunk 3]
```diff
@@ -0,0 +1,41 @@
...
+)
+
+func RunTurbineCmd(appPath string, command TurbineCommand, env map[string]string, flags ...string) (string, error) {
+	cmd := NewTurbineCmd(appPath, command, env, flags...)
+	cmd.Stderr = nil
+	cmd.Stdout = nil
+	cmd.Dir = appPath
+	out, err := cmd.CombinedOutput()
+	return strings.TrimSpace(string(out)), err
+}
```
Comment (dianadoherty) [inline cmd/meroxa/turbine/python/internal/cmd.go:26]: I don't think this is used anywhere

Comment (anna-cross) [reply cmd/meroxa/turbine/python/internal/cmd.go:26]: It is used in version, similarly as we have in Ruby :)

**Pull request closed without merging**